crates/wasi/src/p3/view.rs

```diff
@@ -25,10 +37,20 @@ use crate::p3::ctx::WasiCtx;
 /// [`Store`]: wasmtime::Store
 /// [`Linker`]: wasmtime::component::Linker
 ///
-pub trait WasiView: Send {
+pub trait WasiView: ResourceView + Send {
     /// Yields mutable access to the [`WasiCtx`] configuration used for this
     /// context.
     fn ctx(&mut self) -> &mut WasiCtx;
+
+    #[doc(hidden)]
+    fn as_wasi_impl(&mut self) -> &mut WasiImpl<Self>
+    where
+        Self: Sized,
+    {
+        // TODO: Figure out how to avoid `unsafe`
+        // SAFETY: `WasiImpl` is `repr(transparent)`
+        unsafe { core::mem::transmute(self) }
+    }
```

Comment:
Could you explain a bit more as to why this is necessary?

I forget the matrix of types/traits involved here, but I would naively expect that `&mut WasiImpl<Self>` is basically equivalent to `WasiImpl<&mut Self>`

Reply:
It worked! No `WasiCliImpl` and no need for generics in the host implementations anymore